File: fixtures/javascript-upb05-configuration/transport.js
/** @typedef {Object} TransportCommand
 * @property {bigint} Sequence
 * @property {bigint} Value
 */
export class TransportCommand { constructor(Sequence, Value) { this.Sequence = Sequence; this.Value = Value; } }

/** @typedef {Object} TransportEvent
 * @property {bigint} Sequence
 * @property {bigint} Value
 */
export class TransportEvent { constructor(Sequence, Value) { this.Sequence = Sequence; this.Value = Value; } }

/** @param {TransportCommand} command @returns {TransportEvent} */
export function Dispatch(command) { return new TransportEvent(command.Sequence, BigInt.asIntN(64, command.Value + 1n)); }

/** @param {TransportEvent} event @returns {bigint} */
export function Replay(event) { return event.Value; }

File: fixtures/lua-upb05-configuration/transport.lua
local Seme = assert(_G.Seme, "Seme adapters are required")

---@class TransportCommand
---@field Sequence seme.i64
---@field Value seme.i64

---@class TransportEvent
---@field Sequence seme.i64
---@field Value seme.i64

---@seme-id 8055555555555555555555555555550c
---@param command TransportCommand
---@return TransportEvent
local function Dispatch(command)
  return Seme.record("TransportEvent", { "Sequence", "Value" }, { Sequence = Seme.field(command, "Sequence"), Value = Seme.add(Seme.field(command, "Value"), Seme.i64_literal("1")) })
end

---@seme-id 8055555555555555555555555555550d
---@param event TransportEvent
---@return seme.i64
local function Replay(event)
  return Seme.field(event, "Value")
end

return { Dispatch = Dispatch, Replay = Replay }
